python/ray/serve/controller.py

```diff
@@ -425,6 +426,17 @@ class Checkpoint:
     # TODO(ilr) Rename reconciler to PendingState
 
 
+@dataclass
+class FutureResult:
```

Comment:
As far as I can tell, none of these fields are actually used at the moment. If that's the case, it probably makes more sense to just use an `asyncio.Event` for now and not worry about this dataclass. In general, it's preferable to incrementally introduce the abstractions as you need them instead of trying to drop everything in at the beginning, otherwise you'll end up needing to just refactor this anyways so it's just extra work.

Reply:
My plan was to check these values on the API in a follow-up PR (it would still be an incremental step before the full refactor).